src/askui/agent.py

```diff
@@ -15,6 +16,10 @@
 from .reporting.report import SimpleReportGenerator
 
 
+class InvalidParameterError(Exception):
+    def __init__(self, *args):
+        super().__init__(*args)
```

Comment:
Where do we want to place the exceptions?

Reply:
As close as possible to where they are thrown so this is fine from my perspective for now. But feel free to move it to `scr/askui/exceptions.py` as it is a very generic exception.

Think about extending `ValueError` instead of `Exception` or even using https://docs.pydantic.dev/latest/concepts/validation_decorator  as this provides a very declarative and consistent way for data validation separated from the actual logic of the function.

Reply:
BTW: You can just do (or do you want to prohibit passing kwargs?)

```suggestion
class InvalidParameterError(Exception):
    pass
```

Reply:
I stick to ExeptionException but does not move ti to exceptions.py

Reply:
Okay, but would be good if you check out `ValueError` as this is meant for exactly that kind of exception from my perspective. I don't think we need a custom exception for invalid parameters passed.